source/examples/vulkan/vk_color_cube/vk_color_cube.cc

```diff
@@ -395,28 +394,56 @@ bool AMDVulkanDemo::InitializeGpa()
 #else
     void* gpa_lib = dlopen("libGPUPerfAPIVK.so", RTLD_NOW);
 #endif
+    if (nullptr == gpa_lib)
+    {
+        AMDVulkanDemoVkUtils::Log("ERROR: failed to dlopen the GPA vulkan library");
+        return false;
+    }
 
-    if (nullptr != gpa_lib)
+    GpaGetFuncTablePtrType get_function_table_type = (GpaGetFuncTablePtrType)(dlsym(gpa_lib, "GpaGetFuncTable"));
+    if (nullptr == get_function_table_type)
     {
-        GpaGetFuncTablePtrType get_function_table_type = (GpaGetFuncTablePtrType)(dlsym(gpa_lib, "GpaGetFuncTable"));
-        if (nullptr != get_function_table_type)
-        {
-            GpaFunctionTable* gpa_function_table = new (std::nothrow) GpaFunctionTable();
+        AMDVulkanDemoVkUtils::Log("ERROR: unable to get GpaGetFuncTable pointer from GPA vulkan library");
+        return false;
+    }
 
-            if (nullptr != gpa_function_table)
-            {
-                gpa_function_table->major_version = GPA_FUNCTION_TABLE_MAJOR_VERSION_NUMBER;
-                gpa_function_table->minor_version = GPA_FUNCTION_TABLE_MINOR_VERSION_NUMBER;
-                GpaStatus status                  = get_function_table_type((void*)(gpa_function_table));
+    GpaFunctionTable* gpa_function_table = new (std::nothrow) GpaFunctionTable();
+    if (nullptr == gpa_function_table)
+    {
+        AMDVulkanDemoVkUtils::Log("ERROR: out of memory");
+        return false;
+    }
 
-                if (kGpaStatusOk == status)
-                {
-                    gpu_perf_api_helper_.gpa_function_table_ = gpa_function_table;
-                    return true;
-                }
-            }
-        }
+    gpa_function_table->major_version = GPA_FUNCTION_TABLE_MAJOR_VERSION_NUMBER;
+    gpa_function_table->minor_version = GPA_FUNCTION_TABLE_MINOR_VERSION_NUMBER;
+
+    GpaStatus status = get_function_table_type((void*)(gpa_function_table));
+    if (kGpaStatusOk != status)
+    {
+        AMDVulkanDemoVkUtils::Log("ERROR: GpaGetFuncTable failed with status %d", status);
+        return false;
+    }
```

Comment:
gpa_function_table should be deleted within this conditional to avoid the memory leak.

Reply:
Good catch! Done.